File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int result = 0;
        while (left < right) {
            int water = (right - left) * Math.min(height[left], height[right]);
            result = Math.max(result, water);
            if (height[left] < height[right]) {
              left++;
            } else {
              right--;
            }
        }
        return result;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, h: List[int]) -> int:
        res = 0
        left = 0
        right = len(h) - 1
        while left < right:
            v = min(h[left], h[right]) * (right - left)
            res = max(res, v)
            if h[left] < h[right]:
                left += 1
            else:
                right -= 1
        return res